File: ReadyTasks/CSharp/EulerProject/Task_16/Task_16/Program.cs
using System;
using System.Numerics;

namespace Task_16
{
    class Program
    {
        static int GetPowerSum(int numb, int power)
        {
            BigInteger res = BigInteger.Pow(numb, power);
            int result = 0;
            while (res > BigInteger.Zero)
            {
                result += (int)(res % 10);
                res /= 10;
            }
            return result;
        }
        static void Main(string[] args)
        {
            Console.WriteLine(GetPowerSum(2, 1000));
        }
    }
}

File: ReadyTasks/FSharp/EulerProject/Task_16/Task_16/Program.fs
// Learn more about F# at http://docs.microsoft.com/dotnet/fsharp

open System
open System.Numerics

let rec getDigitSum (numb: bigint) =
    if (numb = 0I) then
        0I
    else
        (numb % 10I) + (getDigitSum (numb / 10I))

[<EntryPoint>]
let main argv =
    printfn "%A" (getDigitSum <| BigInteger.Pow(2I, 1000))
    0 // return an integer exit code